packages/grid/x-data-grid/src/components/toolbar/GridToolbarExport.tsx

```diff
@@ -68,16 +69,32 @@ export const GridPrintExportMenuItem = (props: GridPrintExportMenuItemProps) =>
 
 const GridToolbarExport = React.forwardRef<HTMLButtonElement, GridToolbarExportProps>(
   function GridToolbarExport(props, ref) {
-    const { csvOptions = {}, printOptions = {}, ...other } = props;
+    const { csvOptions = {}, printOptions = {}, excelOptions, ...other } = props;
 
-    if (csvOptions?.disableToolbarButton && printOptions?.disableToolbarButton) {
+    const apiRef = useGridApiContext();
+
+    const defaultButtons = [
+      ...(csvOptions.disableToolbarButton ? [] : [<GridCsvExportMenuItem options={csvOptions} />]),
+      ...(printOptions.disableToolbarButton
+        ? []
+        : [<GridPrintExportMenuItem options={printOptions} />]),
+    ];
+
+    const preProcessedButtons = apiRef.current.unstable_applyPreProcessors(
+      'exportMenu',
+      defaultButtons,
+      excelOptions,
```

Comment:
If we pass `excelOptions` directly, we can't support other buttons in the future.

```suggestion
      { excelOptions },
```